### DISCUSSION TITLE
RFC-7578 - Need API to communicate to expect a file

### DISCUSSION THREAD

**Discussion by YStrauchP4** [encode/starlette#2707] [Q&A]
I have this stack:
`R` client -> `curl` -> FastAPI -> Scarlette.

The client (curl for R) does not allow adding a file name for in-memory uploads. Technically they are correct as RFC-7578 says for this use case, file names are allowed to be omitted:
> The file name isn't mandatory for cases where the file name isn't available or is meaningless or private

So from their perspective, there should not be a need to add a name from their perspective.

Scarlette [docs](https://www.starlette.io/requests/#Request%20Files) says:
> As settled in [RFC-7578: 4.2](https://www.ietf.org/rfc/rfc7578.txt), form-data content part that contains file assumed to have name and filename fields in Content-Disposition header: Content-Disposition: form-data; name="user"; filename="somefile". Though filename field is optional according to RFC-7578, it helps Starlette to differentiate which data should be treated as file. If filename field was supplied, UploadFile object will be created to access underlying file, otherwise form-data part will be parsed and available as a raw string.

I understand the need to somehow distinguish a file upload from a string. But this deliberately ignores what RFC-7578: 4.2 specifies and breaks compatibility with official clients.

FastAPI on the other hand knows that I expect a file because I annotated the parameter. It doesn't have a mechanism to communicate this to Scarlette though, Scarlette guesses the type incorrectly, and it gets rejected by FastAPI (pydantic probably).

I am not deep in the design patterns between FastAPI and Scarlette, but to me it seems that Scarlette needs a way of FastAPI to tell it to expect a file. Is that feasible?
There's another discussion on the FastAPI GitHub where this was raised: 
https://github.com/fastapi/fastapi/discussions/11822

**Comment by rls-p4** [comment]
See related issue/discussion in the `curl` for R repo: https://github.com/jeroen/curl/issues/339
